### ISSUE TITLE
Problem With Export DateTime Variables

### ISSUE THREAD

**Issue by Shoutt** [MySqlBackupNET/MySqlBackup.Net#46]
First, thank you for great job.

I have found an issue exporting DateTime variables. I have studied the library a lite and I have found one solution.

In class `QueryExpress`, the function 
`public static string ConvertToSqlFormat(MySqlDataReader rdr, int colIndex, bool wrapStringWithSingleQuote, bool escapeStringSequence, MySqlColumn col)`
where `else if (ob is System.DateTime)`

You do this to construct the string
`                
                if (wrapStringWithSingleQuote)
                    sb.AppendFormat("'");

                sb.AppendFormat(((DateTime)ob).ToString("yyyy-MM-dd HH:mm:ss", _dateFormatInfo));

                if (col.TimeFractionLength > 0)
                {
                    sb.Append(".");
                    string _microsecond = rdr.GetMySqlDateTime(colIndex).Microsecond.ToString();

                    if (_microsecond.Length < col.TimeFractionLength)
                    {
                        _microsecond = _microsecond.PadLeft(col.TimeFractionLength, '0');
                    }
                    else if (_microsecond.Length > col.TimeFractionLength)
                    {
                        _microsecond = _microsecond.Substring(0, col.TimeFractionLength);
                    }
                    sb.Append(_microsecond.ToString().PadLeft(col.TimeFractionLength, '0'));
                }`

But `rdr.GetMySqlDateTime(colIndex).Microsecond.ToString()` returns a string with lenght equal to number of significant digits, that means:
If we have 0 microseconds, the string will be: "0"
If we have 8900 microseconds, the string will be: "8900"
If we have 384921 microseconds, the string will be: "384921"

So if `col.TimeFractionLength = 3` (miliseconds) and we have 1 milisecond (1000 microseconds) `rdr.GetMySqlDateTime(colIndex).Microsecond.ToString()` returns "1000" and
`                    else if (_microsecond.Length > col.TimeFractionLength)
                    {
                        _microsecond = _microsecond.Substring(0, col.TimeFractionLength);
                    }`
The string will be "100", and we will have convert 1ms to 100ms.

I have fixed the issue with this changes:
`                
                if (wrapStringWithSingleQuote)
                    sb.AppendFormat("'");

                sb.AppendFormat(((DateTime)ob).ToString("yyyy-MM-dd HH:mm:ss", _dateFormatInfo));

                if (col.TimeFractionLength > 0)
                {
                    sb.Append(".");
                    string _microsecond = rdr.GetMySqlDateTime(colIndex).Microsecond.ToString();

                    _microsecond = _microsecond.PadLeft(6, '0');
                    _microsecond = _microsecond.Substring(0, col.TimeFractionLength);
                    sb.Append(_microsecond.ToString().PadLeft(col.TimeFractionLength, '0'));
                }`

**Comment by adriancs2** [comment]
Hey, thanks for fix. After some digging, I find that the .net DateTime will capture the microsecond to 6 digits. So, I have tried the following and it come out to be ok:
```
string _microsecond = ((DateTime)ob).ToString("".PadLeft(col.TimeFractionLength, 'f'));
sb.Append(_microsecond);
```

**Comment by adriancs2** [comment]
Hi, I have released a new version (v2.3), you can try it in your environment and see whether the problem get solved or not.

**Comment by Shoutt** [comment]
Hi, it works, problem solved. Thanks.
